pkg/reader/ccip_test.go

```diff
@@ -1702,25 +1702,46 @@ func TestCCIPChainReader_GetWrappedNativeTokenPriceUSD(t *testing.T) {
 				*pricePtr = price2
 			}).Return(nil)
 
-		ccipReader := &ccipChainReader{
-			destChain: destChain,
-			contractReaders: map[cciptypes.ChainSelector]contractreader.Extended{
+		cw := writer_mocks.NewMockContractWriter(t)
+		contractWriters := make(map[cciptypes.ChainSelector]types.ContractWriter)
+		contractWriters[sourceChain1] = cw
+		contractWriters[sourceChain2] = cw
+
+		ccipReader, err := newCCIPChainReaderInternal(
+			tests.Context(t),
+			logger.Test(t),
+			map[cciptypes.ChainSelector]contractreader.ContractReaderFacade{
 				sourceChain1: sourceReader1,
 				sourceChain2: sourceReader2,
 			},
-			configPoller: mockCache,
-			lggr:         logger.Test(t),
+			contractWriters,
+			destChain,
+			[]byte{0x3}, // Mock offramp address
+			internal.NewMockAddressCodecHex(t),
+		)
+		require.NoError(t, err)
+
+		ccipReader.contractReaders = map[cciptypes.ChainSelector]contractreader.Extended{
+			sourceChain1: sourceReader1,
+			sourceChain2: sourceReader2,
 		}
 
+		// A real cache is still initialized inside of newCCIPChainReaderInternal(). However, in order
+		// to mock it properly, we need to set it to mockCache. Store realCache to close it later.
```

Comment:
Could you elaborate on this? Is it because the constructor creates a real log poller but you want to inject a mocked version?

If that's what's happening, might be worth making a "new cup reader with log poller" constructor.